tests/catboost/test_catboost_model_export.py

```diff
@@ -56,13 +56,30 @@ def read_yaml(path):
     params=[
         cb.CatBoost(MODEL_PARAMS),
         cb.CatBoostClassifier(**MODEL_PARAMS),
+        cb.CatBoostRanker(**MODEL_PARAMS),
         cb.CatBoostRegressor(**MODEL_PARAMS),
     ],
-    ids=["CatBoost", "CatBoostClassifier", "CatBoostRegressor"],
+    ids=["CatBoost", "CatBoostClassifier", "CatBoostRanker", "CatBoostRegressor"],
 )
 def cb_model(request):
     model = request.param
     X, y = get_iris()
+
+    if isinstance(model, CatBoostRanker):
+        # the ranking task requires setting a group_id
+        # we are creating a dummy group_id here that doesn't make any sense for the Iris dataset, but is ok for testing
+        # if the code is running correctly
+
+        indices = np.arange(len(candidates))
+
+        dummy_group_id = np.zeros(len(candidates))
+        dummy_group_id[indices % 2 == 0] = 1  # fill every 2nd value with 1
+        dummy_group_id[indices % 3 == 0] = 2  # fill every 3rd value with 2
+        dummy_group_id = dummy_group_ids.astype("int64")
+        dummy_group_id.sort()
+
+        return ModelWithData(model=model.fit(X, y, group_id=dummy_group_id), inference_dataframe=X)
+
     return ModelWithData(model=model.fit(X, y), inference_dataframe=X)
```

Comment:
```suggestion

    X, y = get_iris()
    return ModelWithData(model=model.fit(X, y), inference_dataframe=X)
```

Can we move `X, y = get_iris()` here?

Reply:
do you want to duplicate this line? 
We need X in order to create the group_id array. Because of that the `X, y = get_iris()` like has to come before the `CatBoostRanker` return.

Reply:
Agree, `candidates` has been replaced with `X`. The current code makes sense.